scripts/scaffold_redux.sh

```diff
@@ -4,62 +4,401 @@
+import 'package:equatable/equatable.dart';
+
+abstract class ${feature_camel_case}State extends Equatable {
+  @override
+  List<Object?> get props => [];
+}
+
 class ${feature_camel_case}NotInitializedState extends ${feature_camel_case}State {}
+
 class ${feature_camel_case}LoadingState extends ${feature_camel_case}State {}
-class ${feature_camel_case}SuccessState extends ${feature_camel_case}State {}
+
 class ${feature_camel_case}FailureState extends ${feature_camel_case}State {}
+
+class ${feature_camel_case}SuccessState extends ${feature_camel_case}State {
+  final bool result;
+
+  ${feature_camel_case}SuccessState(this.result);
+
+  @override
+  List<Object?> get props => [result];
+}
 EOM
 
+
+
 echo "Creating reducer…"
 cat > "lib/features/$feature_snake_case/${feature_snake_case}_reducer.dart" <<- EOM
-import 'package:pass_emploi_app/features/$feature_snake_case/${feature_snake_case}_actions.dart';
+${actionImport}
 import 'package:pass_emploi_app/features/$feature_snake_case/${feature_snake_case}_state.dart';
 
 ${feature_camel_case}State ${feature_first_char_lower_case}Reducer(${feature_camel_case}State current, dynamic action) {
   if (action is ${feature_camel_case}LoadingAction) return ${feature_camel_case}LoadingState();
   if (action is ${feature_camel_case}FailureAction) return ${feature_camel_case}FailureState();
-  if (action is ${feature_camel_case}SuccessAction) return ${feature_camel_case}SuccessState();
+  if (action is ${feature_camel_case}SuccessAction) return ${feature_camel_case}SuccessState(action.result);
   if (action is ${feature_camel_case}ResetAction) return ${feature_camel_case}NotInitializedState();
   return current;
 }
 EOM
 
+
+
 echo "Creating middleware…"
 cat > "lib/features/$feature_snake_case/${feature_snake_case}_middleware.dart" <<- EOM
-import 'package:pass_emploi_app/features/$feature_snake_case/${feature_snake_case}_actions.dart';
-import 'package:pass_emploi_app/features/login/login_state.dart';
+${actionImport}
 import 'package:pass_emploi_app/redux/app_state.dart';
+${repositoryImport}
 import 'package:redux/redux.dart';
 
-class ${feature_camel_case}Middleware extends MiddlewareClass<AppState> {
+class ${middlewareClass} extends MiddlewareClass<AppState> {
+  final ${repositoryClass} _repository;
+
+  ${middlewareClass}(this._repository);
+
   @override
   void call(Store<AppState> store, action, NextDispatcher next) async {
     next(action);
-    final loginState = store.state.loginState;
-    if (loginState is LoginSuccessState && action is ${feature_camel_case}RequestAction) {
+    final userId = store.state.userId();
+    if (userId != null && action is ${feature_camel_case}RequestAction) {
       store.dispatch(${feature_camel_case}LoadingAction());
-      //TODO: call repository
-      //store.dispatch(result != null ? ${feature_camel_case}SuccessAction(action.) : ${feature_camel_case}FailureAction());
+      final result = await _repository.get();
+      if (result != null) {
```

Comment:
Perso j'aime bien le Elvis plutôt ici par défaut, quitte à faire if -else si trop long. As you prefer